src/hrflow_connectors/core/connector.py

```diff
@@ -894,6 +913,7 @@ def based_on(
         connector = cls(
             name=name,
             type=type,
+            subtype=name.lower().replace(" ", "") if subtype is None else subtype,
```

Comment:
Ici je pense que faudrait juste faire `subtype=subtype` parce ici tu duplique la logique qui compute le subtype par default à partir de `name`. 

Si à un moment tu changes de regle tu as du coup deux endroits à modifier. Et tu vas casser the single source of truth. Suffit de passer `subtype` tel quel et laisser la logique dans ton validateur faire le travail